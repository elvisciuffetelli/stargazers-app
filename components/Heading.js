import React from "react";
import { View } from "react-native";
import { Text } from "@rneui/themed";
import PropTypes from "prop-types";
import styles from "../styles";

export default function Heading({ title, subtitle }) {
  return (
    <View style={styles.titleContainer}>
      <Text h1 style={styles.title}>
        {title}
      </Text>
      <Text h3 style={styles.title}>
        {subtitle}
      </Text>
    </View>
  );
}

Heading.defaultProps = {
  title: "",
  subtitle: "",
};

Heading.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
};
