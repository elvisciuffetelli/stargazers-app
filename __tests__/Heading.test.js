import React from "react";
import { render, screen } from '@testing-library/react-native';
import Heading from "../components/Heading";

describe("<Heading />", () => {
  test('It must render the title correctly', async () => {
    const props = {
      title: "Some title",
    };

    render(<Heading
      title={props.title}
    />);

    const title = screen.getByText(props.title);

    expect(title).toBeTruthy();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('It must render the subtitle correctly', async () => {
    const props = {
      subtitle: "Some subtitle",
    };

    render(<Heading
      subtitle={props.subtitle}
    />);

    const subtitle = screen.getByText(props.subtitle);

    expect(subtitle).toBeTruthy();
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
