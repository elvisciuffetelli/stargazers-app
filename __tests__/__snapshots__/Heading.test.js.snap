// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> It must render the subtitle correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#242424",
        "fontSize": 51.42857142857143,
        "fontWeight": "300",
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  />
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#242424",
        "fontSize": 36,
        "fontWeight": "300",
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    Some subtitle
  </Text>
</View>
`;

exports[`<Heading /> It must render the title correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#242424",
        "fontSize": 51.42857142857143,
        "fontWeight": "300",
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    Some title
  </Text>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#242424",
        "fontSize": 36,
        "fontWeight": "300",
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  />
</View>
`;
